### PR TITLE
nperf 1.12.3 (new cask)

### PR DESCRIPTION
**Pull request by rogeriopradoj**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://docs.brew.sh/Acceptable-Casks#finding-a-home-for-your-cask).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/n/nperf.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  depends_on arch: :arm64
+
+  pkg "nPerf-arm64-#{version}.pkg"
+
+  uninstall pkgutil: "com.nperf.nPerf"
+
+  zap trash: [
+    "~/Library/Application Support/nPerf",
+    "~/Library/Saved Application State/com.nperf.nPerf.savedState",
+    "/var/db/receipts/com.nperf.nPerf.bom",
```
Comment (bevanjkay) [inline Casks/n/nperf.rb:24]: These paths are temporary and don't need to be included in `zap`

**Review by bevanjkay — DISMISSED** [review]

File: Casks/n/nperf.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+cask "nperf" do
+  version "1.12.3"
+  sha256 :no_check
+
+  url "https://app.nperf.com/getMacApp?arch=arm64"
```
Comment (bevanjkay) [inline Casks/n/nperf.rb:5]: Is there also an intel version?

**Comment by bevanjkay** [comment]
Rebasing so that this will run the full test suite on the new ARM runners

File: Casks/n/nperf.rb (hunk 2 above)
Comment (Bo98) [reply Casks/n/nperf.rb:5]: There is: https://app.nperf.com/getMacApp?arch=x86_64

Comment (Bo98) [reply Casks/n/nperf.rb:5]: Looks like it redirects to https://repo.nperf.com/macos/nperf/nPerf-x86_64-1.12.3.pkg which is a much better URL that we can version and do sha256 checks on.

Similarly with https://repo.nperf.com/macos/nperf/nPerf-arm64-1.12.3.pkg

File: Casks/n/nperf.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  uninstall pkgutil: "com.nperf.nPerf"
+
+  zap trash: [
+    "~/Library/Application Support/nPerf",
+    "~/Library/Saved Application State/com.nperf.nPerf.savedState",
+    "/var/db/receipts/com.nperf.nPerf.bom",
+    "/var/db/receipts/com.nperf.nPerf.plist",
+  ]
+end
```
[the remark this answers is not in the document]
Comment (miccal) [reply Casks/n/nperf.rb:27]: Thank you @daeho-ro.

File: Casks/n/nperf.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
+cask "nperf" do
+  arch arm: "arm64", intel: "x86_64"
+
+  version "1.12.3"
+  sha256 arm:   "4b2d260a02a1d5567badc2ea15de06c03be80deeb8ef74bb155c5b222b8d4d37",
+         intel: "4aed59b0718a88037a2ab3413e22b10605face4b114010bb3acfb69e83b1f47a"
+
+  url "https://repo.nperf.com/macos/nperf/nPerf-#{arch}-#{version}.pkg"
+  name "nPerf"
+  desc "Test the quality of your Internet connection up to 10 Gb/s"
```
Comment (bevanjkay) [inline Casks/n/nperf.rb:10]:
```suggestion
  desc "Internet speed test utility"

**Review by miccal — APPROVED** [review]

**Review by krehel — APPROVED** [review]

**Pull request merged**